test/e2e/util.go

```diff
@@ -43,26 +48,59 @@ const (
 	defaultTimeout = 30 * time.Second
 )
 
-func restClientConfig(config, context string) (*api.Config, error) {
-	if config == "" {
-		return nil, fmt.Errorf("Config file must be specified to load client config")
-	}
-	c, err := clientcmd.LoadFromFile(config)
-	if err != nil {
-		return nil, fmt.Errorf("error loading config: %v", err.Error())
-	}
-	if context != "" {
-		c.CurrentContext = context
-	}
-	return c, nil
+func CreateTestProject(t *testing.T, kubeClient kubernetes.Interface, projectClient *projectclient.Clientset) string {
+	newNamespace := names.SimpleNameGenerator.GenerateName("e2e-oauth-proxy-")
+
+	// e2e.Logf("Creating project %q", newNamespace)
+	_, err := kubeClient.CoreV1().Namespaces().Create(context.Background(),
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: newNamespace,
+				Labels: map[string]string{
+					"test": "oauth-proxy",
+				},
+			},
+		}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	err = waitForSelfSAR(1*time.Second, 60*time.Second, kubeClient, authorizationv1.SelfSubjectAccessReviewSpec{
+		ResourceAttributes: &authorizationv1.ResourceAttributes{
+			Namespace: newNamespace,
+			Verb:      "create",
+			Group:     "",
+			Resource:  "pods",
+		},
+	})
+	require.NoError(t, err)
+
+	return newNamespace
 }
 
-func loadConfig(config, context string) (*rest.Config, error) {
-	c, err := restClientConfig(config, context)
+func waitForSelfSAR(interval, timeout time.Duration, c kubernetes.Interface, selfSAR authorizationv1.SelfSubjectAccessReviewSpec) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		res, err := c.AuthorizationV1().SelfSubjectAccessReviews().Create(
+			context.Background(),
+			&authorizationv1.SelfSubjectAccessReview{
+				Spec: selfSAR,
+			},
+			metav1.CreateOptions{},
+		)
+		if err != nil {
```

Comment:
do you think `return res.Status.Allowed, err` is sufficient? I am assuming `Allowed` will be set to `false` in case of error.

Reply:
... but `res` will probably be nil. The next condition is unnecessary though and can use what you're suggesting, I'll do that there.